torch/_subclasses/functional_tensor.py

```diff
@@ -547,16 +592,51 @@ def unwrap(x):
                     )
-                    # We don't allow any mutation on result of dropout or _to_copy
-                    if self.export:
-                        if func in (
-                            torch.ops.aten.dropout.default,
-                            torch.ops.aten._to_copy.default,
-                        ):
-                            torch._freeze_functional_tensor(outs_unwrapped)  # type: ignore[attr-defined]
+
                     outs_wrapped = pytree.tree_map_only(
                         torch.Tensor, wrap, outs_unwrapped
                     )
+
+                    if self.export:
+                        # check all args. If any of them is an alias to the result of aten.to and is mutated,
+                        # we mark it in _mutated_partial_frozen_storage.
+                        for arg_id, arg in enumerate(args):
+                            if isinstance(
+                                arg, FunctionalTensor
+                            ) and torch._is_functional_tensor(arg.elem):
+                                arg_storage = StorageWeakRef(arg.untyped_storage())
+                                arg_mutation_counter = torch._mutation_counter(arg.elem)  # type: ignore[attr-defined]
+                                if arg_mutation_counter > arg_mutation_counters[arg_id]:  # type: ignore[index]
+                                    # store tensor aliases that's mutated after it's partial frozen
+                                    arg_group_id = self._partial_frozen_storage.get(
+                                        arg_storage, None
+                                    )
+                                    if arg_group_id is not None:
+                                        self._mutated_partial_frozen_storage.add(
+                                            arg_group_id
+                                        )
+
+                        # TODO: allow it in draft export mode and attach to FailureReport
+                        if func == torch.ops.aten._to_copy.default:
+                            group_id = None
+                            if isinstance(
+                                args[0], FunctionalTensor
+                            ) and torch._is_functional_tensor(args[0].elem):
+                                arg_storage = StorageWeakRef(args[0].untyped_storage())
+                                group_id = self._partial_frozen_storage.get(
+                                    arg_storage, id(arg_storage)
+                                )
+                                self._partial_frozen_storage[arg_storage] = group_id
+                            storage = StorageWeakRef(outs_wrapped.untyped_storage())
+                            if group_id is None:
+                                group_id = id(storage)
+
+                            self._partial_frozen_storage[storage] = group_id
+
+                        # We don't allow any mutation on result of dropout
```

Comment:
Can you include the reasoning in the comment? My understanding is just that we aren't as worried about people running into this in the `dropout` case, and so we don't want to bother implementing it (although to be fair, once you've implemented this logic for `aten.to()` it should be pretty easy to include dropout)

Reply:
@tugsbayasgalan Do we want to do the same thing for dropout as well? I remember we discussed this briefly before and decided not to do it for now, but I forgot why. I think we can give dropout the same treatment?

Reply:
Hmm i think it was just to not increase the blast radius! We can do it as follow up or in this PR, whatever you prefer.

Reply:
sg, I think either is fine (mostly just calling out that the comment makes it unclear to readers whether this is something fundamental or something we want to fix later)